Felienne/hedy
# Classes overview in Teachers Page

- Move classes overview to For Teachers page.
- Hide For Teachers page from main menu unless the user is a teacher.
- Related to (but doesn't close) #546 
- In profile, hides "My classes" for teachers, and puts that section on top of the profile for all other users.
- Improves E2E tests for classes, closes #702 
- Expandable/shrinkable docs (WIP, see below).
- Use syntax highlighting for code snippets in For Teachers page.
- Use a markdown -> HTML pipeline for parsing the markdown docs.
- Load auth messages through the same mechanism that UI & error messages are loaded (`client_messages.js`), to avoid empty modal messages and save load time.
- In class overview, shows link as an actual link (closes #685)
- If a user tries to join a class that they're part of, the page will let them know they're already part of the class.
- Adds rename button to class overview.
- Allows teachers to see list of all programs for students that are part of their classes.
- Note at the top of teachers' profiles that states that the list of their classes has been moved.

**How to test**:
- For Teachers page should not be visible in the heading of either a non-logged in user or a non-teacher user, but it should be visible for users that are teachers.
- If non-teachers go to /for-teachers.html, they can still see the docs but not the table of classes.
- In Profile, teacher users should not be able to see the My Classes section.
- The top of the For Teachers page should contain a table with all the classes owned by the teacher.
- The bottom of the For Teachers page should have a set of docs with sections that can be expanded/shrinked. The code snippets in these docs should be highlighted properly.
- In class overview, shows link as an actual link
- If a user tries to join a class that they're part of, the page will let them know they're already part of the class.
- Rename a class as teacher.
- Be able to see list of programs for each student as a teacher.
- See note at the top of teachers' profile stating that classes have moved.
